model-server/src/test/kotlin/org/modelix/model/server/AuthorizationTest.kt

```diff
@@ -0,0 +1,315 @@
+                      "config": {
+                        "introspection.token.claim": "true",
+                        "claim.value": "[\"model-server/admin\"]",
+                        "userinfo.token.claim": "true",
+                        "id.token.claim": "true",
+                        "lightweight.claim": "false",
+                        "access.token.claim": "true",
+                        "claim.name": "permissions",
+                        "jsonType.label": "JSON",
+                        "access.tokenResponse.claim": "false"
+                      }
+                    }
+                  ]
+                }
+              ],
+              "users": [
+                {
+                  "username": "$USER",
+                  "email": "authorization-test-user@authorization-test-user.com",
+                  "firstName": "authorization-test-user",
+                  "lastName": "authorization-test-user",
+                  "enabled": true,
+                  "credentials": [
+                    {
+                      "type": "password",
+                      "value": "$USER_PASSWORD"
+                    }
+                  ]
+                }
+              ],
+              "components": {
+                "org.keycloak.keys.KeyProvider": [
+                   {
+                    "name": "rsa-256-generated",
+                    "providerId": "rsa-generated",
+                    "subComponents": {},
+                    "config": {
+                      "keySize": [
+                        "2048"
+                      ],
+                      "active": [
+                        "true"
+                      ],
+                      "priority": [
+                        "100"
+                      ],
+                      "enabled": [
+                        "true"
+                      ],
+                      "algorithm": [
+                        "RS256"
+                      ]
+                    }
+                  },
+                  {
+                    "name": "rsa-512-generated",
+                    "providerId": "rsa-generated",
+                    "subComponents": {},
+                    "config": {
+                      "keySize": [
+                        "2048"
+                      ],
+                      "active": [
+                        "true"
+                      ],
+                      "priority": [
+                        "0"
+                      ],
+                      "enabled": [
+                        "true"
+                      ],
+                      "algorithm": [
+                        "RS512"
+                      ]
+                    }
+                  }
+                ]
+              }
+            }
+        """
+
+        // Reuse on container across all tests. The configuration and state does not change in between.
+        private val keycloak: GenericContainer<*> = GenericContainer("quay.io/keycloak/keycloak:25.0.4")
```

Comment:
There would be a distinct testcontainers keycloak module: https://testcontainers.com/modules/keycloak/

Reply:
We can currently not use it, because it does not support Java 11.

See https://github.com/dasniko/testcontainers-keycloak/issues/158

Reply:
wtf, ok :D